app/controllers/api/v1/carts_controller.rb

```diff
@@ -0,0 +1,50 @@
+module Api
+  module V1
+    class CartsController < Api::V1::ApiController
+      before_action :load_cart, :find_product, :cart_item_management,
+                    only: %i[add_product remove_product]
+
+      def add_product
+        @item.add_to_cart
+
+        render json: { message: response_message }, status: :ok
+      end
+
+      def remove_product
+        @item.remove_from_cart
+
+        render json: { message: response_message }, status: :ok
+      end
+
+      private
+
+      def update_cart_params
+        params.require(:cart).permit(:product_id, :quantity)
+      end
+
+      def load_cart
+        @cart = current_user.carts.find_or_create_by!(status: :in_process)
+      end
```

Comment:
why do we need a `find_or_create_by!` method here?

Reply:
This allows the user to always have a `Cart` open in the `in_process` state so that they can add or remove products when logging in to different places.

The state of the `Cart` changes, when in future business logic, it changes to `completed`.